### PR TITLE
# 1521 Attached Template-set plugin for deployment

### LINKED ISSUE #1521: template-set.xml is deployed separately
As a cobigen user, I want to have a performant lookup of available template sets so that I feel comfortable in managing template sets easily. Therefore, information about template-sets should be retrieved without downloading or even extracting a huge amount of data or zip files.

*Acceptance Criteria* 
* [x] The template-set.xml file should be deployed as an additional deployable in the maven process
* [x] The template-set.xml should match the same naming convention like it is automatically done for the pom.xml on deploying to the local or remote m2 repository

Implementation Hint: http://www.mojohaus.org/build-helper-maven-plugin/attach-artifact-mojo.html should do the job easily to add another xml to the deployables.

### PR DESCRIPTION
**Pull request by Ali7223**
Adresses/Fixes #1521
this PR continues and replaces the old [PR](https://github.com/devonfw/cobigen/pull/1559)

Implements

* Implemented attached artifact for each template set
 

@devonfw/cobigen

### CODE REVIEW

**Comment by Ali7223** [comment]
Build does not work because of missing template-set.xml files and the draft is waiting for the files first to get merged.

File: cobigen-templates/crud-angular-client-app/pom.xml [hunk 1]
```diff
@@ -10,6 +10,34 @@
...
+          <execution>
+            <id>attach-artifacts</id>
+            <phase>package</phase>
+            <goals>
+              <goal>attach-artifact</goal>
+            </goals>
+            <configuration>
+              <artifacts>
+                <artifact>
+                  <file>${basedir}\src\main\resources\template-set.xml</file>
```
Comment (jan-vcapgemini) [inline cobigen-templates/crud-angular-client-app/pom.xml:30]: Shouldn't this be in src/main/templates?

**Comment by EduardKrieger** [comment]
Add template-set to the end of the artifact name and relocate the attach process to the templates parent pom

**Review by Zylesto — APPROVED** [review]

**Review by jan-vcapgemini — APPROVED** [review]

**Review by ZakariaeMkassi — APPROVED** [review]

**Comment by Ali7223** [comment]
> Can you also add template-set at the end to the created artifacts. Also the template-set merge will be creating the xml files but i think there is no problem in already adding empty files @jan-vcapgemini ?

'Can you also add template-set at the end to the created artifacts'
Done

**Review by Zylesto — APPROVED** [review]

**Review by EduardKrieger — APPROVED** [review]

**Review by jan-vcapgemini — APPROVED** [review]

**Review by maybeec — CHANGES_REQUESTED** [review]
Strangely I see a lot of emtpy files in the PR.
What is this about?

**Comment by Ali7223** [comment]
> Strangely I see a lot of emtpy files in the PR. What is this about?

I was testing if the project builds and these files were deployed seperately. 

All of the empty files have been removed.

**Comment by jan-vcapgemini** [comment]
> Strangely I see a lot of emtpy files in the PR. What is this about?

I thought they were required because the profile attach-artifacts activation only works if a template-set.xml is existing. The empty files will be filled when the merge ticket is done and the final template-set.xml can be implemented.

**Review by maybeec — APPROVED** [review]

**Pull request merged**